### PR TITLE
[SYCLomatic] Add mapping query of 40 Math APIs.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/test/dpct/asm/asm_v2inst.cu [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  // CHECK-NEXT: d = dpct::extend_vmax2_sat<uint32_t, uint32_t, uint32_t>(a, b, c);
+  // CHECK-NEXT: d = dpct::extend_vmax2_add<uint32_t, uint32_t, uint32_t>(a, b, c);
+  // clang-format off
+  asm("vmax2.u32.u32.u32 %0, %1, %2, %3;" : "=r"(d) : "r"(a), "r"(b), "r"(c));
+  asm("vmax2.u32.u32.u32.sat %0, %1, %2, %3;" : "=r"(d) : "r"(a), "r"(b), "r"(c));
+  asm("vmax2.u32.u32.u32.add %0, %1, %2, %3;" : "=r"(d) : "r"(a), "r"(b), "r"(c));
+  // clang-format on
+}
+
+int main() { return 0; }
```
Comment (yihanwg) [inline clang/test/dpct/asm/asm_v2inst.cu:53]: `main` is not needed.

File: clang/runtime/dpct-rt/include/math.hpp [hunk 2]
```diff
@@ -667,6 +667,13 @@ inline int64_t zero_or_signed_extent(T val, unsigned bit) {
   }
   return val;
 }
+template <typename T>
```
Comment (yihanwg) [inline clang/runtime/dpct-rt/include/math.hpp:670]: Can we reuse `zero_or_signed_extent`? Seems they have same functionality

File: clang/test/dpct/asm/asm_v2inst.cu (hunk 1 above)
Comment (tangjj11) [reply clang/test/dpct/asm/asm_v2inst.cu]: fixed

File: clang/runtime/dpct-rt/include/math.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/math.hpp]: fixed

**Review by yihwang-nv — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**